libswc/shm.c

```diff
@@ -38,6 +38,14 @@
 #include <wld/pixman.h>
 #include <wld/wld.h>
 
+#ifdef __NetBSD__
+	#define swc_mremap(oldp, oldsize, newsize) \
+		mremap((oldp), (oldsize), NULL, (newsize), 0)
+#else /* Linux-style mremap */
+	#define swc_mremap(oldp, oldsize, newsize) \
+		mremap((oldp), (oldsize), (newsize), MREMAP_MAYMOVE)
+#endif
```

Comment:
I think I prefer a regular function over a macro here. But, since it's just a one-off, I think it's fine to just do the `#ifdef` at the usage (as you had previously).

If we build up several little things like this, it's probably worth considering a OS-specific utility file.

Reply:
Any suggestion of where to put it or what to name the file?

Reply:
This is just personal taste at my part, putting any OS-specific code at the top rather than inside functions.

I've turned swc_mremap into a function. I don't think there's enough need for an OS utility file yet, because it would only really contain this one line (unless there's anything else you can spot).

Reply:
That's perfect. Yeah, it's nice to be able to see any OS-specific stuff without having to sift through the whole file.